### PR TITLE
chore(vault): use name instead of prefix to cache vault strategies and schemas

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Just a small change/fix to use right key in strategies/schema caching.

### CODE REVIEW

File: kong/pdk/vault.lua [hunk 1]
```diff
@@ -296,8 +296,8 @@ local function new(self)
 
       schema = schema.fields.config
 
-      STRATEGIES[prefix] = strategy
-      SCHEMAS[prefix] = schema
+      STRATEGIES[name] = strategy
+      SCHEMAS[name] = schema
```
Comment (flrgh) [inline kong/pdk/vault.lua:300]: Asking out of curiosity... what is the problem with this? Were we using `prefix` to index into `STRATEGIES`/`SCHEMAS` elsewhere and thus have a 100% cache miss rate?

Comment (bungle) [reply kong/pdk/vault.lua:300]: @flrgh check this:

https://github.com/Kong/kong/pull/10777/files#diff-212b3488c5eb070820f7bc5178fbaa56e9435fe49cf050feb29091ca9aa1540bR284-R300

It should explain it, especially lines 284 and 285.

Comment (bungle) [reply kong/pdk/vault.lua:300]: The process secrets code is slightly bigger:
https://github.com/Kong/kong/blob/7e4f6ac0a1f3bd3fb157dce70b144d69a52a7b12/kong/pdk/vault.lua#L179-L224

And also contains possible `vault.init` calls (that is where the cache was introduced. In above we could have changed line 284 and 285 to look by `prefix`, but we don't need by prefix here, just by vault name.

**Review by flrgh — APPROVED** [review]
No new test attached, but this seems low risk assuming that we have broader unit tests that exercise both branches (cache hit and miss).

**Comment by kikito** [comment]
Merged, @bungle please cherrypick to EE

**Pull request merged**